src/CppUTestExt/MockSupport_c.cpp

```diff
@@ -311,7 +313,7 @@ MockExpectedCall_c* withConstPointerParameters_c(const char* name, const void* v
 
 MockExpectedCall_c* withFunctionPointerParameters_c(const char* name, void (*value)())
 {
-    expectedCall = &expectedCall->withParameter(name, value);
+    expectedCall = &expectedCall->withParameter(name, (Cpp_Func*)value);
```

Comment:
It would still be a cast to cpp - func. But a c - func is passed via the argument.